lms/djangoapps/instructor_analytics/basic.py

```diff
@@ -462,12 +464,23 @@ def student_responses(course):
             for module in modules:
                 try:
                     state_dict = json.loads(module.state) if module.state else {}
-                    raw_answers = state_dict.get("student_answers", {})
                 except ValueError:
                     log.error("Student responses: Could not parse module state for " +
                               "StudentModule id={module_id}, course={course_id}".format(module_id=module.id, course_id=course.id))
                     continue
-                pretty_answers = u', '.join(u"{problem}={answer}".format(problem=problem, answer=answer) for (problem, answer) in raw_answers.items())
+
+                # Include other problems, e.g. Free Text Response, Submit And Compare Xblocks
+                # that write 'student_answer' to the state.
+                if 'student_answers' in state_dict:
+                    raw_answers = state_dict['student_answers']
+                    pretty_answers = u', '.join(u"{problem}={answer}".format(problem=problem, answer=answer) for (problem, answer) in raw_answers.items())
```

Comment:
This line is a bit long. More than just character count, the combination of the string format along with a list comprehension makes this rather dense and difficult to follow.

This will be easier to read/maintain by splitting it up a bit more, something like:

```python
pretty_answers = u', '.join([
    u"{problem}={answer}".format(
        problem=problem,
        answer=answer,
    )
    for (problem, answer) in raw_answers.items()
])
```